logicmonitor_data_sdk/api/logs.py

```diff
@@ -165,9 +168,26 @@ def _do_request(self):
       self.UnLock()
 
   def _merge_request(self, single_request):
+    # size limiting
+    payload_cache = json.dumps(self._payload_cache)
+    buf_payload_cache = io.BytesIO()
+    with gzip.GzipFile(mode='wb', fileobj=buf_payload_cache) as file:
+      file.write(payload_cache.encode("utf-8"))
+    file.close()
+    compressed_payload_cache = buf_payload_cache.getvalue()
+    serialized_single_request = self.api_client.sanitize_for_serialization(single_request)
+    single_request_json = json.dumps(serialized_single_request)
+    buf_single_request = io.BytesIO()
+    with gzip.GzipFile(mode='wb', fileobj=buf_single_request) as file:
```

Comment:
This is used multiple times, at multiple places. Try to move this to some common place like utils.